### PR TITLE
QuizSessionView, ProgressView, AnswersView 

### PR DESCRIPTION
**Pull request by barbarakos**
Simulator Screen Shot - iPhone 14 Pro - 2022-12-19 at 10 21 15
Added:
- QuizSessionView
- ProgressView
- AnswersView

### CODE REVIEW

File: QuizApp/Quiz/QuizDetailViews/QuizInfoView.swift [hunk 1]
```diff
@@ -63,7 +64,7 @@ struct QuizInfoView_Previews: PreviewProvider {
             difficulty: DifficultyModel.normal,
             imageUrl: "https://upload.wikimedia.org/wikipedia/commons/1/1d/Football_Pallo_valmiina-cropped.jpg",
             name: "Football",
-        numberOfQuestions: 5))
+            numberOfQuestions: 5), startQuiz: {})
```
Comment (marinzg) [inline QuizApp/Quiz/QuizDetailViews/QuizInfoView.swift]: Move `startQuiz` to new line.

File: QuizApp/QuizSession/AnswersView.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                Button {
+                    if answer.index == correctAnswerIndex {
+                        numberOfCorrectQuestions += 1
+                        answers[answer.index].color = .green
+                        nextQuestion(numberOfCorrectQuestions, .green)
+                    } else {
+                        answers[answer.index].color = .red
+                        answers[correctAnswerIndex].color = .green
+                        nextQuestion(numberOfCorrectQuestions, .red)
+                    }
```
Comment (marinzg) [inline QuizApp/QuizSession/AnswersView.swift]: I would move the body of `if` and body of `else` to a new function. SwiftUI is kind of hard to read when colsures are used so having sth like:
```swift
Button {
    if answer.index == correctAnswerIndex {
        correctAnswerSelected()
    } else {
       incorrectAnswerSelected()
    }
}
```

File: QuizApp/QuizSession/QuizSessionView.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    var body: some View {
+        ScrollView {
+            VStack {
+                ProgressView(
+                    progressColors: $progressColors,
+                    numberOfQuestions: viewModel.quiz.numberOfQuestions)
+
+                VStack(alignment: .leading) {
+                    if viewModel.currentQuestion != nil {
```
Comment (marinzg) [inline QuizApp/QuizSession/QuizSessionView.swift]: Why would the current question ever be `nil`?

File: QuizApp/QuizSession/QuizSessionView.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                        Text("\(viewModel.currentQuestion.index+1)/\(viewModel.quiz.numberOfQuestions)")
```
Comment (marinzg) [inline QuizApp/QuizSession/QuizSessionView.swift]: Can the viewModel provide this information?

File: QuizApp/QuizSession/QuizSessionView.swift [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    static var previews: some View {
+        QuizSessionView(viewModel: Container.quizSessionViewModel(QuizModel(id: 0,
+                                                                            category: "SPORT",
+                                                                            description:
+                                                                                "Description of the selected quiz.",
+                                                                            difficulty: DifficultyModel.normal,
+                                                                            imageUrl: "",
+                                                                            name: "Football",
+                                                                            numberOfQuestions: 5)))
```
Comment (marinzg) [inline QuizApp/QuizSession/QuizSessionView.swift]: Indent and set newlines properly.

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/QuizSession/QuizSessionView.swift (hunk 3 above)
Comment (barbarakos) [reply QuizApp/QuizSession/QuizSessionView.swift]: It's ```nil```in the beginning until all the questions are fetched (app crashed without ```if viewModel.currentQuestion != nil```).

File: QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift [hunk 6]
```diff
@@ -1,22 +1,27 @@
 import Combine
+import SwiftUI
 import UIKit
 
-class QuizSessionViewModel {
+class QuizSessionViewModel: ObservableObject {
 
     @Published var currentQuestion: QuestionModel!
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift]: Avoid implicit unwrapping and use optional

File: QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift [hunk 7]
```diff
@@ -1,22 +1,27 @@
...
+    @Published var questionNumberLabel: String = ""
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift]: Does it need to be @Published?

File: QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift [hunk 8]
```diff
@@ -1,22 +1,27 @@
...
     var quiz: QuizModel
     var questions: [QuestionModel] = []
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift]: Can this be private?

File: QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift [hunk 9]
```diff
@@ -68,4 +73,15 @@ class QuizSessionViewModel {
             result: Result(numOfCorrectQuestions: numOfCorrectQuestions, numOfQuestions: questions.count))
     }
 
+    private func subscriptions() {
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift]:
```suggestion
    private func setBindings() {
```

Rename func name to smth actionable

File: QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift [hunk 10]
```diff
@@ -68,4 +73,15 @@ class QuizSessionViewModel {
...
+        $currentQuestion
+            .compactMap { $0 }
+            .sink { [weak self] currentQuestion in
+                guard let self = self else { return }
+
+                self.questionNumberLabel = "\(currentQuestion.index+1)/\(self.quiz.numberOfQuestions)"
```
Comment (dorastipkovic) [inline QuizApp/QuizSession/ViewModel/QuizSessionViewModel.swift]:
```suggestion
                self.questionNumberLabel = "\(currentQuestion.index + 1)/\(self.quiz.numberOfQuestions)"
```

**Review by dorastipkovic — CHANGES_REQUESTED** [review]

**Review by dorastipkovic — APPROVED** [review]

**Pull request merged**